x/coinmaster/bindings/query_plugin.go

```diff
@@ -0,0 +1,44 @@
+package bindings
+
+import (
+	"encoding/json"
+	"fmt"
+
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	coinmastertypes "github.com/noria-net/noria/x/coinmaster/types"
+)
+
+// CustomQuerier dispatches custom CosmWasm bindings queries.
+func CustomQuerier(qp *QueryPlugin) func(ctx sdk.Context, request json.RawMessage) ([]byte, error) {
+	return func(ctx sdk.Context, request json.RawMessage) ([]byte, error) {
+		var contractQuery coinmastertypes.CoinmasterQuery
+		if err := json.Unmarshal(request, &contractQuery); err != nil {
+			return nil, sdkerrors.Wrap(err, "coinmaster query")
+		}
+		if contractQuery.Coinmaster == nil {
+			return nil, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, "nil coinmaster field")
```

Comment:
This differs from the guard check done on [message_plugin.go:42](https://github.com/noria-net/noria/pull/14/files#diff-cf62c38355db48c6113b10f39ae49bedba7e9a8e7b0739b85c5633d6c1caee23R42) - is that expected?

Reply:
The guards are ok because queries are different than messages and do not get dispatched sequencially like messages do. But that comment did make me realize some other issues with the query plugin I wrote (I was today years old when I learned the importance of not setting `omitempty` for wasm bindings!). Fixed in 5c1cd3643ff6c436b6fa4dd33091d8e169668982

Reply:
I too was today years old when I learned what `omitempty` does!